studio/hooks/misc/useCheckPermissions.ts

```diff
@@ -11,18 +12,20 @@ export function useCheckPermissions(
   action: string,
   resource: string,
   data?: object,
-  organizationId?: number
+  // [Joshen] Pass the variables if you want to avoid hooks in this
+  // e.g If you want to use useCheckPermissions in a loop like organization settings
+  organizationId?: number,
+  permissions?: Permission[]
 ) {
-  const { data: permissions } = usePermissionsQuery({
-    enabled: IS_PLATFORM,
-  })
-  const organization = useSelectedOrganization()
-
   if (!IS_PLATFORM) return true
 
-  const orgId = organizationId ?? organization?.id
+  const { data: allPermissions } =
+    permissions === undefined ? usePermissionsQuery() : { data: permissions }
+  const organization =
+    organizationId === undefined ? useSelectedOrganization() : { id: organizationId }
+  const orgId = organization?.id
```

Comment:
these are not following the rules of hooks (eslint is mad lol)

Reply:
hmm this one is tricky - i'm not too sure whats the best fix for this cause it needs to be used conditionally. i've added eslint ignore statements for now but otherwise the other way i'm thinking of is just create a separate method that only does the checking of the permissions (requesting for org id and permissions to be passed, like what this conditional is trying to do)

Reply:
the members dropdown is the only place i think that requires the calling of checking permissions in a loop, so maybe we can create this separate method mentioned in a MembersDropdown.utils or equivalent instead of within this useCheckPermissions